simple/lint.go

```diff
@@ -1839,3 +1840,55 @@ func (c *Checker) LintNilCheckAroundRange(j *lint.Job) {
 		ast.Inspect(f, fn)
 	}
 }
+
+func (c *Checker) LintSort(j *lint.Job) {
+	fn := func(node ast.Node) bool {
+		if !j.IsCallToAST(node, "sort.Sort") {
+			return true
+		}
+		call := node.(*ast.CallExpr)
+		if len(call.Args) == 0 {
```

Comment:
Impossible branch. We know that the `sort.Sort` function has exactly one argument.

Reply:
A code might not compile and nevertheless be syntactically valid. Do we need gosimple to panic on `sort.Sort()`?
`sort` also can be not the stdlib's sort package, but an alias for another package.

Reply:
1. gosimple (as well as staticcheck and my other linters) only operate on code that type checks, i.e. compiles. Ergo, `sort.Sort()` will never reach us.
2. The `sort` in `sort.Sort` is the absolute, resolved import path of the package. It can neither be an alias for another package, nor a vendored, modified copy of `sort`.